pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -78,11 +90,106 @@ func (r *ReconcileStorageCluster) newOBCStorageClass(initData *ocsv1.StorageClus
+	// check for the status boolean value accepted or not
+	if instance.Status.ExternalSecretFound {
+		return nil
+	}
+	found := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: instance.Namespace,
+		},
+	}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: found.Name, Namespace: found.Namespace}, found)
+	if err != nil {
+		return err
+	}
+	ownerRef := metav1.OwnerReference{
+		UID:        instance.UID,
+		APIVersion: instance.APIVersion,
+		Kind:       instance.Kind,
+		Name:       instance.Name,
+	}
+	var data []ocsv1.ExternalResources
+	err = json.Unmarshal(found.Data[externalClusterSecretDataKeyName], &data)
+	if err != nil {
+		return fmt.Errorf("could not parse json blob, got %+v", err)
+	}
+	for _, d := range data {
+		switch d.Kind {
+		case "ConfigMap":
+			cm := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            d.Name,
+					Namespace:       d.Name, // TODO: What will be namespace given for ConfigMap?. d.Name or instance.Namespace?.
+					OwnerReferences: []metav1.OwnerReference{ownerRef},
+				},
+			}
+			cm.Data = d.Data
+			found := &corev1.ConfigMap{}
+			err := r.client.Get(context.TODO(), types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}, found)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					reqLogger.Info("Creating Ceph ConfigMap:	" + cm.Name)
+					err = r.client.Create(context.TODO(), cm)
+					if err != nil {
+						return err
+					}
+				}
+				return err
+			}
+		case "Secret":
+			s := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            d.Name,
+					Namespace:       d.Name, // TODO: What will be namespace for Secret?. d.Name or instance.Namespace?.
+					OwnerReferences: []metav1.OwnerReference{ownerRef},
+				},
+			}
+			for k, v := range d.Data {
+				s.Data[k] = []byte(v)
+			}
+			found := &corev1.Secret{}
+			err := r.client.Get(context.TODO(), types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}, found)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					reqLogger.Info("Creating Ceph Secret: 	" + s.Name)
+					err = r.client.Create(context.TODO(), s)
+					if err != nil {
+						return err
+					}
+				}
+				return err
+			}
+		case "StorageClass":
+			scs, err := r.newStorageClasses(instance)
+			if err != nil {
+				return err
+			}
+			scCephFs := scs[0]
+			scCephFs.Parameters["fsName"] = d.FilesystemName
+			scRBD := scs[1]
+			scRBD.Parameters["pool"] = d.PoolName
+			scOBC := scs[2]
+			scOBC.Parameters["endpoint"] = d.Data["endpoint"] // Setting the OBC StorageClass endpoint
+			err = r.createStorageClasses(scs, reqLogger)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	instance.Status.ExternalSecretFound = true
+	return nil
+}
+
 // newStorageClasses returns the StorageClass instances that should be created
 // on first run.
 func (r *ReconcileStorageCluster) newStorageClasses(initData *ocsv1.StorageCluster) ([]*storagev1.StorageClass, error) {
```

Comment:
Here you want to take the relevant values that we're setting for the external StorageClasses (`fsName`, `pool`, and `endpoint`) and define them above `ret`. Then you'll want a `switch` against the ExternalResources in `initData.Status.ExternalStorageClasses`, setting the appropriate value for each name found.